neuraxle/checkpoints.py

```diff
@@ -163,7 +163,7 @@ def should_resume(self, data_container: DataContainer, context: ExecutionContext
         return True
 
 
-class Checkpoint(NonFittableMixin, NonTransformableMixin, ResumableStepMixin, BaseStep):
+class Checkpoint(NonTransformableMixin, HandleOnlyMixin, ResumableStepMixin, BaseStep):
```

Comment:
Could this simply work?  idea for an issue or future pr. 
```suggestion
class Checkpoint(ResumableStepMixin, TransformerStep):
```

Reply:
I don't think you understand NonTransformableMixin properly. anyways I removed all of the duplicated code. only _did_process is implemented now. added Created DefaultHandlerMethodsMixin to do this.